.github/workflows/scripts/matchRecordToInterface.py

```diff
@@ -0,0 +1,375 @@
+
+submission_file_index = ({ 'file_path': f"{REPORTING_PACKAGE}/SubmissionFileIndex.java", 'record_name': 'SubmissionFileIndex' }, { 'file_path': f"{REPORT_VIEWER_FACTORIES}/ComparisonFactory.ts", 'interface_name': 'ReportFormatSubmissionFileIndex' })
+submission_file = ({ 'file_path': f"{REPORTING_PACKAGE}/SubmissionFile.java", 'record_name': 'SubmissionFile' }, { 'file_path': f"{REPORT_VIEWER_FACTORIES}/ComparisonFactory.ts", 'interface_name': 'ReportSubmissionFile' })
+
+run_information = ({ 'file_path': f"{REPORTING_PACKAGE}/RunInformation.java", 'record_name': 'RunInformation' }, { 'file_path': f"{REPORT_VIEWER_FACTORIES}/RunInformationFactory.ts", 'interface_name': 'ReportFormatRunInformation' })
+failed_submission = ({ 'file_path': f"{REPORTING_PACKAGE}/FailedSubmission.java", 'record_name': 'FailedSubmission' }, { 'file_path': f"{REPORT_VIEWER_MODELS}/RunInformation.ts", 'interface_name': 'FailedSubmission' })
+version = ({ 'file_path': f"{REPORTING_PACKAGE}/Version.java", 'record_name': 'Version' }, { 'file_path': f"{REPORT_VIEWER_FACTORIES}/RunInformationFactory.ts", 'interface_name': 'ReportFormatVersion' })
+
+submission_mappings = ({ 'file_path': f"{REPORTING_PACKAGE}/SubmissionMappings.java", 'record_name': 'SubmissionMappings' }, { 'file_path': f"{REPORT_VIEWER_FACTORIES}/SubmissionMappingsFactory.ts", 'interface_name': 'ReportFormatSubmissionMappings' })
+
+top_comparison = ({ 'file_path': f"{REPORTING_PACKAGE}/TopComparison.java", 'record_name': 'TopComparison' }, { 'file_path': f"{REPORT_VIEWER_FACTORIES}/TopComparisonFactory.ts", 'interface_name': 'ReportFormatTopComparison' })
+
+cluster = ({ 'file_path': f"{REPORTING_PACKAGE}/Cluster.java", 'record_name': 'Cluster' }, { 'file_path': f"{REPORT_VIEWER_MODELS}/Cluster.ts", 'interface_name': 'ReportFormatCluster' })
+
+# we do not check the distribution as it has own record and is just a map directly written to a json file
+
+file_list = [cli_options, merging_options, clustering_options, comparison, match, code_position, basecode_match, submission_file_index, submission_file, run_information, failed_submission, version, submission_mappings, top_comparison, cluster]
+
+typeMatches = {
+    # Basic map from Java types to TypeScript types
+    'int': 'number',
+    'Integer': 'number',
+    'double': 'number',
+    'Double': 'number',
+    'long': 'number',
+    'String': 'string',
+    'boolean': 'boolean',
+    'File': 'string',
+    'SubmissionState': 'SubmissionState',
+    'SimilarityMetric': 'MetricType',
+    'Preprocessing': 'string',
+    'ClusteringAlgorithm': 'string',
+    'InterClusterSimilarity': 'string',
+}
+# add all record interface matches to the typeMatches list
+for java_file, typescript_file in file_list:
+    record_name = java_file['record_name']
+    interface_name = typescript_file['interface_name']
+    typeMatches[record_name] = interface_name
+
+
+def readFileContents(file_path):
+    with open(file_path, 'r') as file:
+        return file.read()
```

Comment:
Should we do exception handling here for cases where we move/rename files? Or is that covered somewhere else?

Reply:
I am a bit confused, as this is the python script here! :D

Reply:
Ah perfect, then this is sufficient, got it!